docs/proof_object.md

```diff
@@ -275,3 +275,91 @@ proof checker extremely simple.
 * Check that `P11` is syntactically equal to `Q1`
 * Check that `P12` is syntactically equal to `P21`
 * Check that `P22` is syntactically equal to `Q2`
+
+# Mixed-level Proof Object Syntax
+
+Language definitions are expected to use mostly
+object-level or mixed-level formulas, which is why
+we have defined the Kore syntax to allow these formulas,
+rather than simply requiring everything to be encoded
+in meta-level formulas.
+For the same reason, we wish to define a syntax of proofs
+that allows writing mixed-level formulas, and that allows
+object-level reasoning to be written in the
+form `by <ProofRule>` rather than by explicitly
+constructing meta-level proofs of terms involving `#provable`
+using the hypotheses described in section 7.7 of
+The Semantics of K.
+
+To support this we allow as patterns any Kore pattern,
+according to the syntax of `<pattern>` in 9.1.4.
+```
+  <ProofObject>  ::= <AssumedClaim>* <DerivedClaim>*
+
+  <AssumedClaim> ::= <Id> : <Pattern>
+
+  <DerivedClaim> ::= <Id> : <Pattern> by <ProofRule>
+
+  <Id> ::= /* Quite flexible syntax. */
+```
+
+Proofs will be defined so that a valid mixed-level proof can
+be transformed into a valid simple proof that assumes or
+proves (resp.) the lifting of each assumed or derived claim
+of the mixed-level proof.
+
+#### Note
+We may wish to disambiguate the object-level proof rule
+
+A pattern is considered equivalent to its lifting.

```

Comment:
Maybe it is worthy to point out that there's a difference between the lifting of a pattern and the one of a pattern as a claim; that is, the lifting process defined in _Section 9.2.8 Lift Patterns_ in _Semantics of K_ lifts a pattern phi to its meta-representation (AST) [[phi]]. The lifting of a claim, say `id : phi`, is `id : #provable([[phi]])`.